Update logging walkthrough (#1462)
defenseunicorns/zarf: docs/13-walkthroughs/3-add-logging.md

> # Add Logging to an Existing Zarf Cluster with Zarf Init

Reviewer: Not a big thing, but this title being longer causes it to wrap @Jessy-Morris not sure if this is a style thing we want to document but this was something beaten into me in the Air Force to avoid (not necessarily saying that is right though)

![image](https://user-images.githubusercontent.com/3977569/227033178-c491e5b1-7644-4e19-b316-08f2069926d8.png)

Author: Made this shorter

## Introduction

In this walkthrough, we are going to show how you can use a Zarf component to inject zero-config, centralized logging into your Zarf cluster.

More specifically, you'll be adding a [Promtail / Loki / Grafana (PLG)](https://github.com/grafana/loki) stack to the [Doom Walkthrough](./2-deploying-doom.md) by installing Zarf's "logging" component.

## System Requirements

- You'll need an internet connection to grab the Zarf Init Package if it's not already on your machine.

## Prerequisites

Prior to this walkthrough you'll want to have a working cluster with Zarf initialized
1.  Zarf binary installed on your $PATH: ([Installing Zarf](../3-getting-started.md#installing-zarf))

> 2. [Initialize a cluster](./1-initializing-a-k8s-cluster.md).

Author: Added

## Walkthrough
[![Walkthrough: Add Logging to an existing Zarf Cluster with Zarf Init](../.images/walkthroughs/logging_thumbnail.png)](https://youtu.be/FYkafs1idlQ "Add Logging to an existing Zarf Cluster with Zarf Init")

1. Run the `zarf init` command on your cluster.

```sh
zarf init
```

2. When prompted to deploy the package select `y` for Yes, then hit the `enter` 
key. <br/> When prompted to deploy the logging component select `y` for Yes, then hit the `enter` key.

> ![The zarf operator is propted to deploy the zarf package and the logging component, selecting 'y' for yes, for each of them.](../.images/walkthroughs/logging_prompts.png)

Reviewer: ```suggestion
![The zarf user is prompted to deploy the zarf package and the logging component, selecting 'y' for yes, for each of them.](../.images/walkthroughs/logging_prompts.png)
```

user is probably better than operator since zarf operator could mean something else to a k8s minded person.

Reviewer: Also can we make this image an iframe too?

4. You can automatically accept the logging component, and confirm the package using the `--confirm` and `--components` flags.

<iframe src="/docs/walkthroughs/logging_init.html" style={{
  height: '800px',
  width: '100%',
  border: 'none'}}></iframe>

### Connecting to the Logging Component

#### Note the Credentials

1. Review the `zarf init` command output for the following:

> ![logging-creds](../.images/walkthroughs/logging_credentials.png)

Reviewer: Can we make this an iframe? (ideally all terminal/code/logging blocks would get converted)

You should see a section for `Logging`.  You will need these credentials later on.

## Deploy the Doom Walkthrough

1. If you haven't already, deploy the [Doom Walkthrough](./2-deploying-doom.md).

## Check the logs

:::note

> Because Doom maybe freshly installed it is recommended to refresh the page a few times to generate more log traffic to view in Grafana

Reviewer: ```suggestion
Because Doom may be freshly installed it is recommended to refresh the page a few times to generate more log traffic to view in Grafana
```

You'll be redirected the `/login` page where you have to sign in with the Grafana credentials you saved [in a previous step](#note-the-credentials).

> ![zarf operator logging into Loki](../.images/walkthroughs/logging_login.png)

Reviewer: ```suggestion
![zarf user logging into Loki](../.images/walkthroughs/logging_login.png)
```

Once you've successfully logged in go to:

1. The "Explore" page (Button on the left that looks like a compass)

> 2. you can select `Loki` in the dropdown, and then

Reviewer: ```suggestion
2. Select `Loki` in the dropdown, and then
```

> 
> 3. enter `{app="game"}` into the Log Browser query input field

Reviewer: ```suggestion
3. Enter `{app="game"}` into the Log Browser query input field
```

> ![zarf operator typing {app='game'} into the Loki Log Browser](../.images/walkthroughs/logging_query.png)

Reviewer: ```suggestion
![zarf user typing {app='game'} into the Loki Log Browser](../.images/walkthroughs/logging_query.png)
```

Submit that query and you'll get back a dump of all the game pod logs that Loki has collected.

> ![Loki query results](../.images/walkthroughs/logging_logs.png)

Reviewer: Same thing here on iframes

## Removal

> 1. Use the `zarf package list` command to get a list of the installed packages and their compoents.  This will give you the name of the init package and the logging component to remove it.

Reviewer: ```suggestion
1. Use the `zarf package list` command to get a list of the installed packages and their components.  This will give you the name of the init package and the logging component to remove it.
```

<iframe src="/docs/walkthroughs/logging_list.html" width="100%"></iframe>

2. Use the `zarf package remove` command to remove the `logging` component from the init package.  Don't forget the `--confirm` flag.  Otherwise you'll receive an error.

<iframe src="/docs/walkthroughs/logging_remove.html" width="100%" height="120px"></iframe>

## Troubleshooting

> ### Unable to connect to the Kubernetes cluster, or timed out waiting for cluster to report healthy.

Reviewer: Same thing about hanging chad wrapping here
![image](https://user-images.githubusercontent.com/3977569/227033347-33c0d77e-a2a8-485a-9b58-b179ece62b13.png)

Author: Made this shorter
